detekt-psi-utils/src/test/kotlin/dev/detekt/psi/AnnotationExcluderSpec.kt

```diff
@@ -1,51 +1,37 @@
 package dev.detekt.psi
 
-import dev.detekt.test.createBindingContext
-import dev.detekt.test.utils.KotlinCoreEnvironmentTest
-import dev.detekt.test.utils.KotlinEnvironmentContainer
+import dev.detekt.test.utils.KotlinAnalysisApiEngine
 import dev.detekt.test.utils.compileContentForTest
 import org.assertj.core.api.Assertions.assertThat
 import org.jetbrains.kotlin.psi.KtAnnotationEntry
 import org.jetbrains.kotlin.psi.KtFile
 import org.jetbrains.kotlin.psi.KtFunction
-import org.jetbrains.kotlin.resolve.BindingContext
 import org.junit.jupiter.api.Disabled
 import org.junit.jupiter.api.Nested
 import org.junit.jupiter.api.Test
 import org.junit.jupiter.params.ParameterizedTest
 import org.junit.jupiter.params.provider.CsvFileSource
 
-@KotlinCoreEnvironmentTest
-class AnnotationExcluderSpec(private val env: KotlinEnvironmentContainer) {
-    private val annotationsKtFile = compileContentForTest(
-        """
-            package dagger
-            
-            annotation class Component {
-                annotation class Factory
-            }
-        """.trimIndent()
-    )
+class AnnotationExcluderSpec {
 
     @ParameterizedTest(
-        name = "Given {0} is excluded when the {1} is found then the excluder returns {2} without type solving"
+        name = "Given {0} is excluded when the {1} is found then the excluder returns {2} without Analysis API"
     )
     @CsvFileSource(resources = ["/annotation_excluder.csv"])
     fun `all cases`(exclusion: String, annotation: String, shouldExclude: Boolean) {
-        val (file, ktAnnotation) = createKtFile(annotation)
-        val excluder = AnnotationExcluder(file, listOf(exclusion.toRegex()), BindingContext.EMPTY)
+        val (file, ktAnnotation) = createKtFile(annotation, false)
```

Comment:
```suggestion
        val (file, ktAnnotation) = createKtFile(annotation, analysisApi = false)
```
Best to update this throughout the tests, I was confused what it meant. The parameter name could be more descriptive too, or maybe there should be a new function `createKtFileWithAA` instead.

Reply:
Done.